### PR TITLE
chore(lint/useShorthandFunctionType): ignore if node has inner comments

### LINKED ISSUE #354: 📎 Implement `lint/useShorthandFunctionType` - `typescript-eslint/prefer-function-type`
### Description

[prefer-function-type](https://typescript-eslint.io/rules/prefer-function-type/)

Want to contribute? Lets us know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by seitarof**
## Summary
close #354

Does not report and fix if node has inner comments https://github.com/biomejs/biome/pull/670#pullrequestreview-1727244683

## Test Plan
Add snaps for valid cases

### CODE REVIEW

**Comment by seitarof** [comment]
I've implemented it to ignore reports and fixes, but should I at least report it?

File: crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_function_type.rs [hunk 1]
```diff
@@ -95,6 +95,10 @@ impl Rule for UseShorthandFunctionType {
 
     fn run(ctx: &RuleContext<Self>) -> Self::Signals {
         let query = ctx.query();
+        // If there are comments, it's not a single call signature.
+        if query.syntax().has_comments_direct() {
+            return None;
+        }
```
Comment (Conaclos) [inline crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_function_type.rs:101]: I think we should still report cases with inner comments, however, we should not suggest any fixes. Thus, this check should be moved in `action`.

**Comment by Conaclos** [comment]
Thanks :)

File: crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_function_type.rs (hunk 1 above)
Comment (seitarof) [reply crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_function_type.rs]: thanks for comment.
fix 797c48b

**Pull request merged**